pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -0,0 +1,573 @@
+		if failureMessage == "" {
+			logrus.Infof("waiting for cluster EKS [%s] to update", cluster.Name)
+			return e.setUnknown(cluster, apimgmtv3.ClusterConditionUpdated, "")
+		}
+		logrus.Infof("waiting for cluster EKS [%s] update failure to be resolved", cluster.Name)
+		return e.setFalse(cluster, apimgmtv3.ClusterConditionUpdated, failureMessage)
+	default:
+		if cluster.Spec.EKSConfig.Imported {
+			cluster, err = e.setUnknown(cluster, apimgmtv3.ClusterConditionPending, "")
+			if err != nil {
+				return cluster, err
+			}
+			logrus.Infof("waiting for cluster import [%s] to start", cluster.Name)
+		} else {
+			logrus.Infof("waiting for cluster create [%s] to start", cluster.Name)
+		}
+		e.clusterEnqueueAfter(cluster.Name, 20*time.Second)
+		return cluster, nil
+	}
+}
+
+func (e *eksOperatorController) updateEKSClusterConfig(cluster *mgmtv3.Cluster, eksClusterConfigDynamic *unstructured.Unstructured, oldSpec map[string]interface{}) (*mgmtv3.Cluster, error) {
+	// configs are not equal, need to update EKS cluster config
+	// getting resource version for watch
+	list, err := e.dynamicClient.Namespace(namespace.GlobalNamespace).List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		return cluster, err
+	}
+	selector := fields.OneTermEqualSelector("metadata.name", cluster.Name)
+	w, err := e.dynamicClient.Namespace(namespace.GlobalNamespace).Watch(context.TODO(), v1.ListOptions{ResourceVersion: list.GetResourceVersion(), FieldSelector: selector.String()})
+	if err != nil {
+		return cluster, err
+	}
+	eksClusterConfigDynamic.Object["spec"] = oldSpec
+	eksClusterConfigDynamic, err = e.dynamicClient.Namespace(namespace.GlobalNamespace).Update(context.TODO(), eksClusterConfigDynamic, v1.UpdateOptions{})
+	if err != nil {
+		return cluster, err
+	}
+
+	// update applied spec
+	cluster, err = e.recordAppliedSpec(cluster)
+	if err != nil {
+		return cluster, err
+	}
+
+	// EKS cluster and node group statuses are not always immediately updated. This cause the EKSConfig to
+	// stay in "active" for a few seconds, causing the cluster to go back to "active".
+	timeout := time.NewTimer(10 * time.Second)
+	for {
+		select {
+		case event := <-w.ResultChan():
+			eksClusterConfigDynamic = event.Object.(*unstructured.Unstructured)
+			status, _ := eksClusterConfigDynamic.Object["status"].(map[string]interface{})
+			if status["phase"] != "updating" {
+				continue
+			}
+
+			// this enqueue is necessary to ensure that the controller is reentered with the updating phase
+			e.clusterEnqueueAfter(cluster.Name, 10*time.Second)
+			return e.setUnknown(cluster, apimgmtv3.ClusterConditionUpdated, "")
+		case <-timeout.C:
+			return cluster, nil
+		}
+	}
+}
+func (e *eksOperatorController) generateAndSetServiceAccount(cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
+	// service account token and API endpoint are need to deploy cluster agent, should be able to retrieve
+	// from secret
+	for i := 1; i < 15; i++ {
```

Comment:
use the kubernetes backoff, start at like .5 seconds and it can still max at 15*20 seconds.